### PR TITLE
added drag and drop functionality

### PR DESCRIPTION
**Pull request by refael-m**
https://cloudinary.atlassian.net/browse/APPS-6163

### CODE REVIEW

File: src/commons/BrowserUtils.ts [hunk 1]
```diff
@@ -868,6 +868,22 @@ export namespace BrowserUtils {
...
 
+  /**
+   * Drag an item to a destination element or position.
+   *
+   * Validate element is visible before drag and drop it
+   * @param selector element selector
+   * @param target destination element selector or object with x and y properties
+   */
+  export function dragAndDrop(selector: string, target: string | WebdriverIO.DragAndDropCoordinate): void {
+    Reporter.debug(`Drag and drop element '${selector}' to ${target}.`);
```
[the remark this answers is not in the document]
Comment (refael-m) [reply src/commons/BrowserUtils.ts:879]: done.

File: src/commons/TestUtils.ts [hunk 2]
```diff
@@ -19,4 +19,12 @@ export namespace TestUtils {
       )
       .join('');
   }
+
+  /**
+   * Return string that contains only the numbers from the original string.
+   * @param str string to get only numbers from
+   */
+  export function getOnlyNumbersFromString(str: string): string {
```
[the remark this answers is not in the document]
Comment (refael-m) [reply src/commons/TestUtils.ts:27]: done.

File: src/commons/BrowserUtils.ts [hunk 3]
```diff
@@ -868,6 +868,22 @@ export namespace BrowserUtils {
...
+    waitForEnabled(selector);
+    tryBlock(
+      () => $(selector).dragAndDrop(typeof target === 'string' ? $(target) : target),
```
[the remark this answers is not in the document]
Comment (refael-m) [reply src/commons/BrowserUtils.ts]: added an existence check for the target.

File: src/commons/BrowserUtils.ts [hunk 4]
```diff
@@ -868,6 +869,28 @@ export namespace BrowserUtils {
...
+   * @param selector element selector
+   * @param target destination element selector or object with x and y properties
+   */
+  export function dragAndDrop(selector: string, target: string | WebdriverIO.DragAndDropCoordinate): void {
+    Reporter.debug(`Drag and drop element '${selector}' to ${inspect(target)}.`);
+    const isTargetSelector: boolean = typeof target === 'string';
+
+    waitForDisplayed(selector);
+    if (isTargetSelector) {
+      isExist(target as string);
```
Comment (FelixZilber) [inline src/commons/BrowserUtils.ts:885]: what the purpose of "as string"?

**Review by FelixZilber — APPROVED** [review]

**Pull request merged**